app/models/resource.rb

```diff
@@ -52,6 +52,12 @@ def find_if_visible role, *a
       res = find *a
       res if res.try :visible_to?, role
     end
+
+    # Specialization to allow lookup by composite ids,
+    # eg. Resource[account, kind, id]
```

Comment:
Just a note for later (not specific to your code), but just to make you aware: we have similar logic to this in multiple places in our code base (and others I believe) including my rotator code.  At some point we should DRY it all up.

Reply:
Agreed. That's why I added this method in lieu of copying the same functionality again, even though I didn't go the one step further and found all the places where it could be used.

Reply:
I don't understand you what you mean, this does copy the same functionality again.  It can't be reused beause here it exists in a very specific context (inside an audit event object), whereas in the codebase in general it's used in many different contexts.  I'm not saying this because I want you to change anything atm, just want to clarify my initial point.

Reply:
What prevents this method from being used in all the other different contexts?